src/Xamarin.Android.Build.Tasks/Utilities/ZipArchiveEx.cs

```diff
@@ -110,6 +111,34 @@ public void AddDirectory (string folder, string folderInArchive, CompressionMeth
 			}
 		}
 
+		/// <summary>
+		/// HACK: aapt2 is creating zip entries on Windows such as `assets\subfolder/asset2.txt`
+		/// </summary>
+		public void FixupWindowsPathSeparators (Action<string> callback)
+		{
+			var streams = new List<MemoryStream> ();
+			foreach (var entry in zip) {
+				//HACK: Since FullName is "fixed up" already, we can tell this is a broken file if ContainsEntry returns false
+				if (!zip.ContainsEntry (entry.FullName)) {
+					callback?.Invoke (entry.FullName);
+
+					var stream = new MemoryStream ();
+					entry.Extract (stream);
+					zip.DeleteEntry (entry);
+
+					stream.Seek (0, SeekOrigin.Begin);
+					zip.AddEntry (entry.FullName, stream, entry.CompressionMethod);
+					streams.Add (stream);
+				}
+			}
+			if (streams.Count > 0) {
+				Flush ();
+				foreach (var stream in streams) {
+					stream.Dispose ();
+				}
+			}
+		}
```

Comment:
@grendello what I'm doing here is bad... Maybe you have a suggestion?

Here is what is happening:
- `packaged_resources` contains a file named `assets\subfolder/asset2.txt`
- `ZipEntry.FullName` returns `assets/subfolder/asset2.txt`
- `ZipArchive.ContainsEntry (entry.FullName)` returns `false` on this entry

There might be a bug here, since the behavior seems a little weird.

Either way, I need to change the path of this entry, and not change anything else. Is there an API in libZipSharp to do that? I don't want to touch a `Stream`.

Reply:
@jonathanpeppers Please test https://github.com/grendello/LibZipSharp/commit/de1712c85c4c016ca3e18ef372563aae723d7a1f 
Note that `.FullName` will now contain unaltered (directory separator char wise) form of the entry's name and there's a new `.NativeFullName` property which has the OS-specific form of the name. Also, added `ZipEntry.Rename`